Mathlib/Order/Category/SemilatCat.lean

```diff
@@ -0,0 +1,215 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+
+! This file was ported from Lean 3 source module order.category.Semilat
+! leanprover-community/mathlib commit e8ac6315bcfcbaf2d19a046719c3b553206dac75
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Order.Category.PartOrdCat
+import Mathlib.Order.Hom.Lattice
+
+/-!
+# The categories of semilattices
+
+This defines `SemilatSup` and `SemilatInf`, the categories of sup-semilattices with a bottom
+element and inf-semilattices with a top element.
+
+## References
+
+* [nLab, *semilattice*](https://ncatlab.org/nlab/show/semilattice)
+-/
+
+set_option linter.uppercaseLean3 false
+
+universe u
+
+open CategoryTheory
+
+/-- The category of sup-semilattices with a bottom element. -/
+structure SemilatSup : Type (u + 1) where
```

Comment:
Trivial doc-strings are okay: they still appear in pop-ups and hovers in situations where otherwise context might be missing, and the trivial statement is still helpful.

Reply:
Yes, let's put `Cat` on the names for consistency.